### PR TITLE
[zh-cn] sync pods/pod-lifecycle.md

### PR DESCRIPTION
**Pull request by my-git9**
content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md

### CODE REVIEW

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 1]
```diff
@@ -294,18 +294,46 @@ the `Terminated` state.
...
+the Pod level `restartPolicy` is either `OnFailure` or `Always`.
 -->
 ## 容器重启策略 {#restart-policy}
 
 Pod 的 `spec` 中包含一个 `restartPolicy` 字段，其可能取值包括
 Always、OnFailure 和 Never。默认值是 Always。
 
-`restartPolicy` 适用于 Pod 中的所有容器。`restartPolicy` 仅针对同一节点上
+`restartPolicy` 适用于 Pod
+中的{{< glossary_tooltip text="应用容器" term_id="app-container" >}}容器和常规的
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:311]:
```suggestion
中的{{< glossary_tooltip text="应用容器" term_id="app-container" >}}和常规的
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 2]
```diff
@@ -294,18 +294,46 @@ the `Terminated` state.
...
+[init 容器](/zh-cn/docs/concepts/workloads/pods/init-containers/)。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:312]:
```suggestion
[Init 容器](/zh-cn/docs/concepts/workloads/pods/init-containers/)。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 3]
```diff
@@ -294,18 +294,46 @@ the `Terminated` state.
...
+[Sidecar 容器](/zh-cn/docs/concepts/workloads/pods/sidecar-containers/)忽略
+Pod 级别的 `restartPolicy` 字段：在 Kubernetes 中，Sidecar 被定义为
+`initContainers` 内的一个条目，其容器级别的 `restartPolicy` 被设置为 `Always`。
+对于因错误而退出的 init 容器，如果 Pod 级别 `restartPolicy` 为 `OnFailure` 或 `Always`，
+则 kubelet 会重新启动 init 容器。
```
Comment (windsonsea) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:317]:
```suggestion
对于因错误而退出的 Init 容器，如果 Pod 级别 `restartPolicy` 为 `OnFailure` 或 `Always`，
则 kubelet 会重新启动 Init 容器。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 4]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+{{<note>}}
+<!--
+Beginning with Kubernetes 1.29, if your Pod includes one or more sidecar containers
+(init containers with an Always restart policy), the kubelet will delay sending
+the TERM signal to these sidecar containers until the last main container has fully terminated.
+The sidecar containers will be terminated in the reverse order they are defined in the Pod spec.
+This ensures that sidecar containers continue serving the other containers in the Pod until they are no longer needed.
+-->
+从 Kubernetes 1.29 开始，如果你的 Pod 包含一个或多个 Sidecar
+容器（具有重启策略为 `Always` 的 Init 容器），kubelet 将延迟向这些
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1022]:
```suggestion
容器（重启策略为 `Always` 的 Init 容器），kubelet 将延迟向这些
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 5]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+Sidecar 容器发送 TERM 信号，直到最后一个主容器完全终止。
+Sidecar 容器将以 Pod 规范中定义的相反顺序终止。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1024]:
```suggestion
Sidecar 容器将以 Pod 规约中定义的相反顺序终止。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 6]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+请注意，主容器的缓慢终止也会延迟边车容器的终止。
+如果宽限期在终止过程完成之前到期，Pod 可能会进入紧急终止状态。
+在这种情况下，Pod 中的所有剩余容器将在短暂的宽限期内同时终止。
+
+<!--
+Similarly, if the Pod has a preStop hook that exceeds the termination grace period, emergency termination may occur.
+In general, if you have used preStop hooks to control the termination order without sidecar containers, you can now
+remove them and allow the kubelet to manage sidecar termination automatically.
+-->
+同样，如果 Pod 的 preStop hook 超过了终止宽限期，则可能会发生紧急终止。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1041]:
```suggestion
同样，如果 Pod 的 preStop 回调超过了终止宽限期，则可能会发生紧急终止。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 7]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+一般来说，如果你在没有 Sidecar 容器的情况下使用 preStop hook 来控制终止顺序，
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1042]:
```suggestion
一般来说，如果你在没有 Sidecar 容器的情况下使用 preStop 回调来控制终止顺序，
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 8]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+那么现在可以删除它们并允许 kubelet 自动管理 Sidecar 终止。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1043]:
```suggestion
那么现在可以删除它们从而允许 kubelet 自动管理 Sidecar 终止。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 9]
```diff
@@ -294,18 +294,46 @@ the `Terminated` state.
...
+kubelet resets the restart backoff timer for that container.
+[Sidecar containers and Pod lifecycle](/docs/concepts/workloads/pods/sidecar-containers/#sidecar-containers-and-pod-lifecycle)
+explains the behaviour of `init containers` when specify `restartpolicy` field on it.
+-->
+当 kubelet 根据配置的重启策略处理容器重启时，仅适用于同一 Pod
+内替换容器并在同一节点上运行的重启。当 Pod 中的容器退出时，`kubelet`
+会按指数回退方式计算重启的延迟（10s、20s、40s、...），其最长延迟为 5 分钟。
+一旦某容器执行了 10 分钟并且没有出现问题，`kubelet` 对该容器的重启回退计时器执行重置操作。
+[Sidecar 容器和 Pod 生命周期](/zh-cn/docs/concepts/workloads/pods/sidecar-containers/#sidecar-containers-and-pod-lifecycle)中解释了
+`init containers` 在指定 `restartpolicy` 字段时的行为。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:334]:
```suggestion
当 kubelet 根据配置的重启策略处理容器重启时，仅适用于在同一节点上同一 Pod 内
创建替代容器的重启。在 Pod 中的容器退出后，`kubelet`
会以指数退避延迟（10s、20s、40s、...）重启它们，该延迟最多五分钟。
一旦容器正常执行了 10 分钟，`kubelet` 会重置该容器的重启退避计时器。
[边车容器和 Pod 生命周期](/zh-cn/docs/concepts/workloads/pods/sidecar-containers/#sidecar-containers-and-pod-lifecycle)解释了在其上指定 `restartpolicy` 字段时 `init containers` 的行为。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 10]
```diff
@@ -294,18 +294,46 @@ the `Terminated` state.
...
+
+`restartPolicy` 仅针对同一节点上
 `kubelet` 的容器重启动作。当 Pod 中的容器退出时，`kubelet`
 会按指数回退方式计算重启的延迟（10s、20s、40s、...），其最长延迟为 5 分钟。
 一旦某容器执行了 10 分钟并且没有出现问题，`kubelet` 对该容器的重启回退计时器执行重置操作。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:339]: please remove these lines

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 11]
```diff
@@ -464,26 +492,28 @@ When a Pod's containers are Ready but at least one custom condition is missing o
...
-在 Pod 被调度到某节点后，它需要被 Kubelet 接受并且挂载所需的卷。
+在 Pod 被调度到某节点后，它需要被 kubelet 接受并且挂载所需的存储卷。
 一旦这些阶段完成，Kubelet 将与容器运行时（使用{{< glossary_tooltip term_id="cri" >}}）
 一起为 Pod 生成运行时沙箱并配置网络。如果启用了 `PodReadyToStartContainersCondition` 
-[特性门控](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/)，
-kubelet 会通过 Pod 的 `status.conditions` 字段中的 `PodReadyToStartContainers` 状况来报告
-Pod 是否达到了初始化里程碑。
+[特性门控](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/)
+（Kubernetes {{< skew currentVersion >}} 版本中默认启用），
+`PodReadyToStartContainers` 状况会被加到 Pod 的 `status.conditions` 字段中。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:516]:
```suggestion
`PodReadyToStartContainers` 状况会被添加到 Pod 的 `status.conditions` 字段中。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 12]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+(init containers with an Always restart policy), the kubelet will delay sending
+the TERM signal to these sidecar containers until the last main container has fully terminated.
+The sidecar containers will be terminated in the reverse order they are defined in the Pod spec.
+This ensures that sidecar containers continue serving the other containers in the Pod until they are no longer needed.
+-->
+从 Kubernetes 1.29 开始，如果你的 Pod 包含一个或多个 Sidecar
+容器（重启策略为 `Always` 的 Init 容器），kubelet 将延迟向这些
+Sidecar 容器发送 TERM 信号，直到最后一个主容器完全终止。
+Sidecar 容器将以 Pod 规约中定义的相反顺序终止。
+这可确保 Sidecar 容器继续为 Pod 中的其他容器提供服务，直到不再需要它们为止。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1025]:
```suggestion
从 Kubernetes 1.29 开始，如果你的 Pod 包含一个或多个边车
容器（重启策略为 `Always` 的 Init 容器），kubelet 将延迟向这些
边车容器发送 TERM 信号，直到最后一个主容器完全终止。
边车容器将以 Pod 规约中定义的相反顺序终止。
这可确保边车容器继续为 Pod 中的其他容器提供服务，直到不再需要它们为止。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 13]
```diff
@@ -980,6 +1010,39 @@ feature gate `EndpointSliceTerminatingCondition` is enabled.
...
+在这种情况下，Pod 中的所有剩余容器将在短暂的宽限期内同时终止。
+
+<!--
+Similarly, if the Pod has a preStop hook that exceeds the termination grace period, emergency termination may occur.
+In general, if you have used preStop hooks to control the termination order without sidecar containers, you can now
+remove them and allow the kubelet to manage sidecar termination automatically.
+-->
+同样，如果 Pod 的 preStop 回调超过了终止宽限期，则可能会发生紧急终止。
+一般来说，如果你在没有 Sidecar 容器的情况下使用 preStop 回调来控制终止顺序，
+那么现在可以删除它们从而允许 kubelet 自动管理 Sidecar 终止。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1043]:
```suggestion
同样，如果 Pod 的 preStop 回调超过了终止宽限期，则可能会发生紧急终止。
一般来说，如果你在没有边车容器的情况下使用 preStop 回调来控制终止顺序，
那么现在可以删除它们从而允许 kubelet 自动管理边车容器终止。
```

File: content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md [hunk 14]
```diff
@@ -1128,12 +1191,15 @@ for more details.
...
+* Learn more about [sidecar containers](/docs/concepts/workloads/pods/sidecar-containers/).
+
 * For detailed information about Pod and container status in the API, see
   the API reference documentation covering
   [`status`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodStatus) for Pod.
 -->
 * 动手实践[为容器生命周期时间关联处理程序](/zh-cn/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/)。
 * 动手实践[配置存活态、就绪态和启动探针](/zh-cn/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)。
 * 进一步了解[容器生命周期回调](/zh-cn/docs/concepts/containers/container-lifecycle-hooks/)。
+* 进一步了解 [Sidecar 容器](/zh-cn/docs/concepts/workloads/pods/sidecar-containers/)。
```
Comment (chenrui333) [inline content/zh-cn/docs/concepts/workloads/pods/pod-lifecycle.md:1203]:
```suggestion
* 进一步了解 [边车容器](/zh-cn/docs/concepts/workloads/pods/sidecar-containers/)。
```

**Review by chenrui333 — COMMENTED** [review]
left couple of comments

**Pull request merged**